cell2location/models/base/_pyro_mixin.py

```diff
@@ -494,3 +497,103 @@ def plot_QC(self, summary_name: str = "means", use_n_obs: int = 1000):
+        loss_fn: Optional[pyro.infer.ELBO] = None,
+        optim: Optional[pyro.optim.PyroOptim] = None,
+        optim_kwargs: Optional[dict] = None,
+        n_aggressive_epochs: int = 20,
+        n_aggressive_steps: int = 100,
+        n_steps_kl_warmup: Union[int, None] = None,
+        n_epochs_kl_warmup: Union[int, None] = 400,
+    ):
+        super().__init__(
+            pyro_module=pyro_module,
+            loss_fn=loss_fn,
+            optim=optim,
+            optim_kwargs=optim_kwargs,
+            n_steps_kl_warmup=n_steps_kl_warmup,
+            n_epochs_kl_warmup=n_epochs_kl_warmup,
+        )
+
+        self.n_aggressive_epochs = n_aggressive_epochs
+        self.n_aggressive_steps = n_aggressive_steps
+        self.n_aggressive_steps_total = n_aggressive_epochs * n_aggressive_steps
+        self.aggressive_steps_counter = 0
+        self.aggressive_total_counter = 0
+
+        self.svi_nonamortised = pyro.infer.SVI(
+            model=pyro.poutine.block(self.pyro_model, hide=list(self.module.list_obs_plate_vars["sites"].keys())),
+            guide=pyro.poutine.block(self.pyro_guide, hide=list(self.module.list_obs_plate_vars["sites"].keys())),
+            optim=self.optim,
+            loss=self.loss_fn,
+        )
+
+        self.svi_amortised = pyro.infer.SVI(
+            model=pyro.poutine.block(self.pyro_model, expose=list(self.module.list_obs_plate_vars["sites"].keys())),
+            guide=pyro.poutine.block(self.pyro_guide, expose=list(self.module.list_obs_plate_vars["sites"].keys())),
+            optim=self.optim,
+            loss=self.loss_fn,
+        )
+
+        self.svi = pyro.infer.SVI(
+            model=self.pyro_model,
+            guide=self.pyro_guide,
+            optim=self.optim,
+            loss=self.loss_fn,
+        )
+
+    def training_step(self, batch, batch_idx):
+
+        args, kwargs = self.module._get_fn_args_from_batch(batch)
+        # Set KL weight if necessary.
+        # Note: if applied, ELBO loss in progress bar is the effective KL annealed loss, not the true ELBO.
+        if self.use_kl_weight:
+            kwargs.update({"kl_weight": self.kl_weight})
+
+        if self.aggressive_total_counter <= self.n_aggressive_steps_total:
+            self.aggressive_total_counter += 1
+            if self.aggressive_steps_counter <= self.n_aggressive_steps:
+                self.aggressive_steps_counter += 1
+                # Do parameter update exclusively for amortised variables
+                loss = torch.Tensor([self.svi_amortised.step(*args, **kwargs)])
+            else:
+                self.aggressive_steps_counter = 0
+                # Do parameter update exclusively for non-amortised variables
+                loss = torch.Tensor([self.svi_nonamortised.step(*args, **kwargs)])
+        else:
+            # Do parameter update for both types of variables
+            loss = torch.Tensor([self.svi.step(*args, **kwargs)])
+
+        return {"loss": loss}
```

Comment:
I am wondering if this is the correct way to do aggressive updating of amortised variables as described here: https://arxiv.org/pdf/1901.05534.pdf

This approach heuristically sets the number of inner and outer loop training settings rather than using mutual information as described in the paper.